src/utils/p5Utils.js

```diff
@@ -33,75 +43,81 @@ class P5Util {
 
   /**
    * This function will check if the p5 is in the environment
-   * Either it is in the p5Instance mode OR it is in the window 
-   * @returns {boolean} if it is in p5 
+   * Either it is in the p5Instance mode OR it is in the window
+   * @returns {boolean} if it is in p5
    */
   checkP5() {
     return !!this.p5Instance;
   }
 
   /**
-     * Convert a canvas to Blob
-     * @param {HTMLCanvasElement} inputCanvas 
-     * @returns {Blob} blob object
-     */
+   * Convert a canvas to a Blob object.
+   * @param {HTMLCanvasElement} inputCanvas
+   * @returns {Promise<Blob>}
+   */
   /* eslint class-methods-use-this: ["error", { "exceptMethods": ["getBlob"] }] */
   getBlob(inputCanvas) {
-    return new Promise((resolve) => {
+    return new Promise((resolve, reject) => {
       inputCanvas.toBlob((blob) => {
-        resolve(blob);
+        if (blob) {
+          resolve(blob);
+        } else {
+          reject(new Error('Canvas could not be converted to Blob.'));
+        }
       });
     });
   };
 
   /**
-     * Load image in async way.
-     * @param {String} url 
-     */
+   * Load a p5.Image from a URL in an async way.
+   * @param {string} url
+   * @return {Promise<p5.Image>}
+   */
   loadAsync(url) {
-    return new Promise((resolve) => {
+    return new Promise((resolve, reject) => {
       this.p5Instance.loadImage(url, (img) => {
         resolve(img);
+      }, () => {
+        reject(new Error(`Could not load image from url ${url}`));
       });
     });
   };
 
   /**
-     * convert raw bytes to blob object
-     * @param {Array} raws 
-     * @param {number} x 
-     * @param {number} y 
-     * @returns {Blob}
-     */
-  async rawToBlob(raws, x, y) {
+   * convert raw bytes to blob object
+   * @param {number[] | Uint8ClampedArray | ArrayLike<number>} raws
+   * @param {number} width
+   * @param {number} height
+   * @returns {Promise<Blob>}
+   */
+  async rawToBlob(raws, width, height) {
```

Comment:
positive affirmation:
* this is so helpful and makes much more sense to use width & height 🙌